manifests/director/console.pp

```diff
@@ -0,0 +1,58 @@
+# @summary Define a Bacula Director Console
+#
+# This define creates a console declaration for the director.
+# Resources of this type are intended to manage conf.d/console.conf entries.
+#
+# @param conf_dir    The bacula configuration director.  Should not need adjusting
+# @param password    The password that must be supplied for a named Bacula Console to be authorized
+# @param jobacl      A list of Job resource names that can be accessed by the console.
+# @param clientacl   A list of Client resource names that can be accessed by the console.
+# @param storageacl  A list of Storage resource names that can be accessed by the console.
+# @param scheduleacl A list of Schedule resource names that can be accessed by the console.
+# @param poolacl     A list of Pool resource names that can be accessed by the console.
+# @param filesetacl  A list of FileSet resource names that can be accessed by the console.
+# @param catalogacl  A list of Catalog resource names that can be accessed by the console.
+# @param commandacl  A list of of console commands that can be executed by the console.
+# @param whereacl    This directive permits you to specify where a restricted console can restore files.
+#
+# Aside from Director resource names and console command names,
+# the special keyword *all* can be specified in any of the above access control lists.
+# When this keyword is present, any resource or command name (which ever is appropriate) will be accepted.
+#
+# @example
+#   bacula::director::console { 'Monitoring':
+#     password     => 'monitoring_password',
+#   }
+#
+define bacula::director::console (
+  String $conf_dir              = $bacula::conf_dir,
+  String $password              = 'some_weak_password',
+  Optional[String] $jobacl      = undef,
+  Optional[String] $clientacl   = undef,
+  Optional[String] $storageacl  = undef,
+  Optional[String] $scheduleacl = undef,
+  Optional[String] $poolacl     = undef,
+  Optional[String] $filesetacl  = undef,
+  String $catalogacl            = '*all*',
+  String $commandacl            = 'list',
```

Comment:
Regarding `$commandacl`, can we enhance validation with something like:

```
Array[Enum['run','list',...]] $commandacl            = ['list'],
```

And in the template `$commandacl.join(', ')`?

Reply:
There is list of Commands and Special dot Commands on https://www.bacula.org/15.0.x-manuals/en/console/Bacula_Enterprise_Console.html#blb:ConsoleChapter.
Do you really want enum of all of them?

Reply:
I guess these `@` commands are out-of-scope, for the other, if it can help avoid misconfiguration it seems appropriate.  But if you think this will be more of a bore than something that can help find misconfiguration before it happen, we can ignore it.

Reply:
Check it pls.

Reply:
:100: